### PR TITLE
Render html pages with javascript in test requests

### PR DESCRIPTION
**Pull request by oflatt**
Now using the requests_html library to get pages with any html elements that are inserted by javascript code.
Also, a refactor with testing making submissions public.

### CODE REVIEW

**Comment by cjmyers** [comment]
Should this be merged?

**Comment by oflatt** [comment]
Yes, but the travis build is still not running.
How do we fix this again, @zachzundel ?

**Comment by cjmyers** [comment]
Not sure if this helps, but I see this error message:

Could not find a version that satisfies the requirement requests-html==0.10.0 (from -r tests/test_requirements.txt (line 6)) (from versions: 0.0.1, 0.1.0, 0.1.1, 0.2.0, 0.2.1, 0.2.2)
No matching distribution found for requests-html==0.10.0 (from -r tests/test_requirements.txt (line 6))
The command "sudo pip install -r tests/test_requirements.txt" failed and exited with 1 during .

**Comment by oflatt** [comment]
That's the error I was trying to fix before travis stopped running on the branch. I may or may not have fixed it by upgrading pip, but I can't get it to run.

**Comment by cjmyers** [comment]
Hmm, my guess is that it has to do with your changes to the .travis.yml file.  Could there be a syntax error or something in this file?

**Comment by zachzundel** [comment]
If Travis is just outright refusing to run, it’s probably a syntax error in
the .travis.yml file

Zach

File: .travis.yml [hunk 1]
```diff
@@ -4,17 +4,21 @@ language: node_js
...
+
 services:
     - docker
 
 jobs:
     include:
         - stage: test
-          before_install:
-              - sudo apt-get install -y python3 python3-pip
+	  before_install:
```
Comment (cjmyers) [inline .travis.yml]: This should be spaces and not a tab.

**Comment by oflatt** [comment]
Thank you! I've been trying to figure out why it's invalid. How did you figure it out? Is there a validator?

**Comment by cjmyers** [comment]
Zach looked at it and noticed in two seconds :-)

**Comment by oflatt** [comment]
Sorry to bother you @zachzundel , but do you know why it can't find the 0.10.0 requests-html package? I'm running the same version of Python and I can install it via pip.

**Pull request merged**